packages/shared/src/components/post/collection/CollectionPostContent.tsx

```diff
@@ -0,0 +1,159 @@
+  enableShowShareNewComment,
+  origin,
+  position,
+  inlineActions,
+  onPreviousPost,
+  onNextPost,
+  onClose,
+  postPosition,
+  isFallback,
+  customNavigation,
+  onRemovePost,
+  backToSquad,
+}: PostContentProps): ReactElement => {
+  const { subject } = useToastNotification();
+  const engagementActions = usePostContent({
+    origin,
+    post,
+  });
+  const { createdAt, contentHtml, image } = post;
+  const { onSharePost: onShare, onReadArticle } = engagementActions;
+
+  const hasNavigation = !!onPreviousPost || !!onNextPost;
+  const containerClass = classNames(
+    'tablet:pb-0 tablet:flex-row',
+    className?.container,
+  );
+
+  const navigationProps: PostNavigationProps = {
+    postPosition,
+    onPreviousPost,
+    onNextPost,
+    post,
+    onReadArticle,
+    onClose,
+    onShare,
+    inlineActions,
+    onRemovePost,
+  };
+
+  const date = useMemo(
+    () => createdAt && postDateFormat(createdAt),
+    [createdAt],
+  );
+
+  return (
+    <PostContentContainer
+      hasNavigation={hasNavigation}
+      className={containerClass}
+      aria-live={subject === ToastSubject.PostContent ? 'polite' : 'off'}
+    >
+      {position === 'fixed' && (
+        <FixedPostNavigation
+          {...navigationProps}
+          className={className?.fixedNavigation}
+        />
+      )}
+
+      <PostContainer
+        className={classNames('relative gap-8', className?.content)}
+        data-testid="postContainer"
+      >
+        {!hasNavigation && (
+          <CollectionPostHeaderActions
+            onShare={onShare}
+            post={post}
+            onClose={onClose}
+            className="flex tablet:hidden mb-4"
+            contextMenuId="post-widgets-context"
+          />
+        )}
+
+        <BasePostContent
+          className={{
+            ...className,
+            onboarding: classNames(
+              className?.onboarding,
+              backToSquad && 'mb-6',
+            ),
+            navigation: {
+              actions: className?.navigation?.actions,
+              container: classNames('pt-6', className?.navigation?.container),
+            },
+          }}
+          isFallback={isFallback}
+          customNavigation={customNavigation}
+          enableShowShareNewComment={enableShowShareNewComment}
+          shouldOnboardAuthor={shouldOnboardAuthor}
+          navigationProps={navigationProps}
+          engagementProps={engagementActions}
+          origin={origin}
+          post={post}
+        >
+          {/* TODO: replace <div>Collection</div> with the right pill */}
+          <div>Collection</div>
+
+          <h1
+            className="font-bold break-words typo-large-title"
+            data-testid="post-modal-title"
+          >
+            {post.title}
+          </h1>
+          <div className="flex items-center text-theme-label-tertiary typo-callout">
+            <span>Last updated</span> <Separator />
+            {/* TODO: change this with metadataChangedAt */}
```

Comment:
This needs to be addressed in this PR.

Reply:
API implementation here https://github.com/dailydotdev/daily-api/pull/1555